### PR TITLE
Add ability to validate layouts

### PR DESCRIPTION
**Pull request by 14rcole**
Causes layouts to be validated when `linchpin up` is run.  As of now, layouts cannot be validated with `linchpin validate`, but I can add that to this PR if #647 is merged first, otherwise I'll submit a separate PR once that is merged

### CODE REVIEW

**Comment by herlo** [comment]
@14rcole can you rebase this against current develop so I may review? Thanks!

**Comment by 14rcole** [comment]
@herlo will do

**Comment by herlo** [comment]
[test]

**Comment by 14rcole** [comment]
@herlo PTAL

File: docs/source/includes/validate.rst [hunk 1]
```diff
@@ -1,6 +1,8 @@
-The purpose of the validate command is to determine whether topologies are syntactically valid and, if not, provide a list of errors that occured during validation
+Validate Command
+-----------------
 
-Using the validate command
+The purpose of the validate command is to determine whether topologies and layouts are syntactically valid.  If not, it will provide a list of errors that occured during validation
+>>>>>>> Add ability to validate layouts:docs/source/advanced/validator.rst
```
Comment (herlo) [inline docs/source/includes/validate.rst]: I'm supposing the `>>>>>>>` is left over from a git conflict?

File: linchpin/__init__.py [hunk 2]
```diff
@@ -473,6 +473,28 @@ def _gen_error_msg(self, prefix, section, error):
...
+        """
+        Validate the provided layout against the schema
+
+        :param layout: layout dictionary
+        """
+
+        pb_path = self._find_playbook_path("layout")
+        try:
+            sp = "{0}/roles/common/files/schema.json".format(pb_path)
+            print sp
```
Comment (herlo) [inline linchpin/__init__.py]: This print should probably not be there.

File: linchpin/__init__.py [hunk 3]
```diff
@@ -688,6 +709,14 @@ def do_action(self, provision_data, action='up', run_id=None, tx_id=None):
 
             if provision_data[target].get('layout', None):
                 l_data = provision_data[target]['layout']
+                try:
+                    self._validate_layout(l_data)
+                except SchemaError as e:
+                    print e
+                    print "\n"
```
Comment (herlo) [inline linchpin/__init__.py]: These two print statements should either be in a ctx.log_state, or managed somehow else.

**Comment by 14rcole** [comment]
[test]

**Comment by herlo** [comment]
[test]

File: docs/source/includes/validate.rst [hunk 4]
```diff
@@ -1,6 +1,7 @@
-The purpose of the validate command is to determine whether topologies are syntactically valid and, if not, provide a list of errors that occured during validation
+Validate Command
+-----------------
 
-Using the validate command
+The purpose of the validate command is to determine whether topologies and layouts are syntactically valid.  If not, it will provide a list of errors that occured during validation
 ```````````````````````````
```
Comment (herlo) [inline docs/source/includes/validate.rst]: If you leave the backticks in here, it will attempt to make the lines above it a title.

**Comment by 14rcole** [comment]
[merge]

**Pull request merged**